File: Courseware/qbxt/第一天/day1-1/1.cpp
cmath pow(x,y)

y>>1 = y/2

int ksm(int x,int y,int p)
{
	if (y==0) return 1;
	int z = ksm(x,y>>1,p);
	z = 1ll*z*z%p;
	if (y%2==1) z=1ll*z*x%p;
	return z;
}
int ksm(int x,int y,int p)
{
	int ans=1;
	while (y)
	{
		if (y&1) ans=1ll*ans*x%p;
		x=1ll*x*x%p;
		y>>=1;
	}
	return ans;
}

File: JavaScriptArea/1.js
function X2(a, b, c, d) {
    let n = a + b + c + d;
    return n * (a * d - b * c) * (a * d - b * c) / ((a + b) * (c + d) * (a + c) * (b + d));
}
function jc (x) {
    let ans = 1;
    for (let i = 1; i <= x; i++) ans = ans * i;
    return ans;
}
function C(n, m) { return jc(n) / (jc(m) * jc(n-m));}
console.log(X2(64, 16, 10, 10))